modules/queue/queue_redis.go

```diff
@@ -83,7 +86,7 @@ func NewRedisByteFIFO(config RedisByteFIFOConfiguration) (*RedisByteFIFO, error)
 		queueName: config.QueueName,
 	}
 	fifo.client = nosql.GetManager().GetRedisClient(config.ConnectionString)
-	if err := fifo.client.Ping().Err(); err != nil {
+	if err := fifo.client.Ping(graceful.GetManager().HammerContext()).Err(); err != nil {
```

Comment:
unfortunately I don't think this is the right context for this one.

Reply:
looking forward to good suggestions ;)

Reply:
OK thinking on this should be ShutdownContext()

```suggestion
	if err := fifo.client.Ping(graceful.GetManager().ShutdownContext()).Err(); err != nil {
```

That is because you're opening a redis queue here and if you're trying to shutdown you want it to shutdown right now.